File: wk9/java/photoshootForGorillas.java
// problem link: https://codeforces.com/contest/2000/problem/E
// solution link: https://codeforces.com/contest/2000/submission/288175896

import java.io.*;
import java.math.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(r.readLine());
        while(t-- > 0) {
            long[] nmk = Arrays.stream(r.readLine().split(" ")).mapToLong(Long::parseLong).toArray();
            int w = Integer.parseInt(r.readLine());
            long[] a = Arrays.stream(r.readLine().split(" ")).mapToLong(Long::parseLong).toArray();
            Arrays.sort(a);
            PriorityQueue<Long> pq = new PriorityQueue<>((x, y) -> -Long.compare(x, y));
            for(int i = 0; i < nmk[0]; i++)
                for(int j = 0; j < nmk[1]; j++)
                    pq.add((Math.min(i, nmk[0] - nmk[2]) - Math.max(-1, i - nmk[2])) * (Math.min(j, nmk[1] - nmk[2]) - Math.max(-1, j - nmk[2])));
            long ret = 0;
            for(int i = a.length - 1; i > -1; i--)
                ret += a[i] * pq.poll();
            System.out.println(ret);
        }
    }
}

File: wk9/cpp/photoshootForGorillas.cpp
// problem link: https://codeforces.com/contest/2000/problem/E
// solution link: https://codeforces.com/contest/2000/submission/295424332

#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    
    int t;
    cin >> t;
    while(t--) {
        int n, m, k, w;
        cin >> n >> m >> k >> w;
        vector<long long> a(w);
        while(w--)
            cin >> a[w];
        sort(a.begin(), a.end());
        priority_queue<long long> pq;
        for(int i = 0; i < n; i++)
            for(int j = 0; j < m; j++)
                pq.push((min(i, n - k) - max(-1, i - k)) * (min(j, m - k) - max(-1, j - k)));
        long long ret = 0;
        
        for(int i = a.size() - 1; i > -1; i--, pq.pop())
            ret += a[i] * pq.top();
        cout << ret << '\n';
    }
        
    return 0;
}